### PR TITLE
test: add NUTs

### PR DESCRIPTION
**Pull request by mdonnalley**
### What does this PR do?
- Adds NUTs for `auth:list`, `auth:logout`, `auth:jwt:grant`, and `auth:sfdxurl:store`

This PR does NOT add any tests for `auth:web:login` or `auth:device:login` since we do not want to mock interactions with a web browser

### What issues does this PR fix or reference?
@W-8856588@

Depends on https://github.com/salesforcecli/cli-plugins-testkit/pull/32

### CODE REVIEW

File: test/commands/auth/logout.nut.ts
Comment (WillieRuemmele) [inline test/commands/auth/logout.nut.ts:25]: I _believe_ that we want to avoid creating too many `TestSessions` and just reuse one and reset/restore it with commands instead of creating new ones... But Steve would know for sure. If you're testing a login flow, call logout in the beforeEach, instead of creating a new testSession

File: test/commands/auth/jwt/grant.nut.ts
Comment (WillieRuemmele) [inline test/commands/auth/jwt/grant.nut.ts:34]: but is this validating JSON, human readable output for us? If no, then we need to verify that

File: test/commands/auth/sfdxurl/store.nut.ts
Comment (WillieRuemmele) [inline test/commands/auth/sfdxurl/store.nut.ts:37]: same comment about TestSession doing JWT for us as above

**Review by WillieRuemmele — CHANGES_REQUESTED** [review]

**Review by WillieRuemmele — APPROVED** [review]

**Review by mshanemc — CHANGES_REQUESTED** [review]

File: test/commands/auth/list.nut.ts
Comment (mshanemc) [reply test/commands/auth/list.nut.ts:49]: not a required change

File: test/commands/auth/jwt/grant.nut.ts
Comment (amphro) [inline test/commands/auth/jwt/grant.nut.ts:50]: Do we want this to be hardcoded? What if the org changes? Can it also be an envar?

File: test/commands/auth/jwt/grant.nut.ts
Comment (amphro) [inline test/commands/auth/jwt/grant.nut.ts:53]: Same with this. What if we want to target this nut on a prod instance?

Comment (amphro) [reply test/commands/auth/jwt/grant.nut.ts:53]: I agree it is a little weird to have asserted values in env vars, but hardcoded doesn't seem right either. You could create a core connection, get the instance URL, and query for the org ID before asserting. I kinda like that solution because it validates a valid token after the exchange.

File: test/commands/auth/logout.nut.ts
Comment (mdonnalley) [reply test/commands/auth/logout.nut.ts:59]: @mshanemc why would it be so bad to include config as dev plugin? It doesn't make any difference to users and it ensures that we have the commands we need during testing.

I like the dev plugin approach more than the `shelljs` approach you're proposing because I'd rather us _always_ use `execCmd` as opposed to sometimes doing one thing sometimes doing another thing

File: test/commands/auth/list.nut.ts
Comment (amphro) [inline test/commands/auth/list.nut.ts:37]: Ditto on all of this.

File: test/commands/auth/list.nut.ts
Comment (amphro) [reply test/commands/auth/list.nut.ts:49]: I agree.

File: test/commands/auth/logout.nut.ts
Comment (amphro) [inline test/commands/auth/logout.nut.ts:64]: Do we actually need to validate that the config was removed, rather than just a success?

File: test/commands/auth/logout.nut.ts
Comment (amphro) [inline test/commands/auth/logout.nut.ts:41]: can we verify that it doesn't show up in auth:list anymore, or that the file is removed?

**Review by amphro — CHANGES_REQUESTED** [review]

File: test/commands/auth/jwt/grant.nut.ts
Comment (mdonnalley) [reply test/commands/auth/jwt/grant.nut.ts:53]: @amphro what if we didn't check the values and instead just checked that those properties existed? After all, plugin-auth isn't responsible for creating the auth info

File: test/commands/auth/jwt/grant.nut.ts
Comment (mshanemc) [reply test/commands/auth/jwt/grant.nut.ts:50]: on plugin-user, I did mine asserting the key and the length.

File: test/commands/auth/jwt/grant.nut.ts
Comment (mshanemc) [reply test/commands/auth/jwt/grant.nut.ts:53]: ex: https://github.com/salesforcecli/plugin-user/blob/7cf495ec31a3ce847d38b14ea8c93b8167e37397/test/allCommands.nut.ts#L30

Comment (amphro) [reply test/commands/auth/jwt/grant.nut.ts:53]: That's fine with me. I just don't want it hardcoded.

File: test/commands/auth/logout.nut.ts
Comment (mshanemc) [reply test/commands/auth/logout.nut.ts:65]: OK, you've got me partially convinced.

My concerns were cross-dependencies (ex: two that end up depending on each other) and the experience we have now of (bump core, bump plugin, bump cli, lockfiles everywhere) and introducing extra steps to get everybody updated, especially if a plugin has a major version (or coreV3).

The "give me sfdx latest" seems easier.

**Review by mshanemc — APPROVED** [review]

File: test/commands/auth/jwt/grant.nut.ts
Comment (peternhale) [inline test/commands/auth/jwt/grant.nut.ts:58]: @mdonnalley this expectation seems to point at a specific dev hub org Id. If dev hub used in for this test changes then test will fail. Maybe just check `Successfully authorized ${username}`

File: test/commands/auth/sfdxurl/store.nut.ts
Comment (peternhale) [inline test/commands/auth/sfdxurl/store.nut.ts:58]: @mdonnalley please remove org id from expectation.

File: test/testHelper.ts
Comment (peternhale) [inline test/testHelper.ts:39]: @mdonnalley local builds are moving to https soon, but not quite there yet. This would break on my local build today. Maybe `/^https*:\/\//.test(auth[urlKey]` instead of startsWith??

**Review by peternhale — CHANGES_REQUESTED** [review]
Very nice @mdonnalley!

File: test/commands/auth/logout.nut.ts
Comment (mdonnalley) [reply test/commands/auth/logout.nut.ts:65]: @mshanemc I think relying on an update to sfdx-cli@latest would be more frustrating than relying on an update to a plugin. For example, what happens if plugin-auth nut test requires a fix in plugin-config? If we rely on latest, then we have to wait for the change to be pulled in during the release which only happens once a week - and then what if there's a moratorium? Then you're blocked until the following week.

If we rely just on the plugin, we can bump the version as soon as the fix is merged in. I personally feel that that's easier

**Review by peternhale — APPROVED** [review]

File: test/commands/auth/logout.nut.ts
Comment (amphro) [reply test/commands/auth/logout.nut.ts:65]: Ya, I agree with Mike here. Also, it is only running on the dependency when targeting `./bin/run`. When the tests are run against `sfdx` (any version) it will use the config command from **that** CLI version which seems correct.

**Review by amphro — APPROVED** [review]

File: test/commands/auth/jwt/grant.nut.ts
Comment (shetzel) [inline test/commands/auth/jwt/grant.nut.ts:35]: nit: I think `await testSession?.clean();` is a good pattern here because if something goes haywire with test session creation and it's not defined, this will fail too and it's just noise in the test results.

**Review by shetzel — APPROVED** [review]
This is awesome!  🎉

**Pull request merged**
